README.md

```diff
@@ -1,16 +1,658 @@
+|---------------|----------------------------------------------|-----------|--------------------------------------------------|----------|---------|
+| `ucan/invoke` | `CID`                                        |           | The CID of the UCAN to invoke                    | Yes      |         |
+| `v`           | `SemVer`                                     | `"0.1.0"` | SemVer of the UCAN invocation this schema        | Yes      |         |
+| `run`         | `"*" or {String: {URI : {Ability : [Any]}}}` |           | Which UCAN capabilities to run                   | No       | `"*"`   |
+| `nnc`         | `String`                                     |           | A unique nonce, to distinguish each invocation   | Yes      |         |
+| `ext`         | `Any`                                        |           | Non-normative extended fields                    | No       | `null`  |
+| `sig`         | `Bytes`                                      |           | Signature of the rest of the field canonicalized | Yes      |         |
+  
+### 3.1.1 Invocation
+
+The `ucan/invoke` field MUST contain CIDs pointing to the UCANs to invoke. The outmost UCAN being invoked SHOULD NOT contain any actions that are not intended to be executed.
+
+### 3.1.2 Version
+
+The `v` field MUST contain the version of the invocation object  schema.
+
+### 3.1.3 Run Capabilities
+
+The OPTIONAL `run` field MUST reference the actions contained in the UCAN are to be run during the invocation. To run all actions in the underlying UCAN, the `"*"` value MUST be used. If only specific actions (or [pipelines](#5-promise-pipelining)) are intended to be run, then they MUST be prefixed with an arbitrary label and treated as a UCAN attenuation: all actions MUST be backed by a matching capability of equal or lesser scope.
+
+#### 3.1.3.1 Promises
+
+The only difference from general capabilities is that [promises](#5-promise-pipelining) MAY also be used as inputs to attenuated fields.
+
+If a capability input has the key `"_"` and the value is a promise, the input MUST be discarded and only used for determining sequencing actions.
+
+### 3.1.4 Nonce
+
+The `nnc` field MUS contain a unique nonce. This field makes each invocation unique.
+
+### 3.1.5 Extended Fields
+
+The OPTIONAL `ext` field MAY contain arbitrary data. If not present, the `ext` field MUST me interpreted as `null`, including for [signature](#315-signature).
+
+### 3.1.6 Signature
+
+The `sig` field MUST contain the signature of the other fields. To construct the payload, the other fields MUST first be canonicalized as `dag-cbor`.
+
+If serialized as JSON, the `sig` field MUST be serialized as [unpadded base64url](https://datatracker.ietf.org/doc/html/rfc4648#section-5).
+
+## 3.2 IPLD Schema
+
+``` ipldsch
+type Invocation struct {
+  inv &UCAN  -- The UCAN providing authority
```

Comment:
The rest of the spec uses `ucan/invoke` here. I'm assuming it should be consistent, and we just missed this here?

Reply:
My theory is that because it's a constant value, we don't need it inside the IPLD. But I could be entirely wrong. @Gozala any sage advice?

Reply:
(Context: In Haskell I would omit this from the data model and add it at serialization time)

Reply:
I think @matheus23 was asking why this called `inv` here as opposed to `ucan/invoke`. I don't think IPLD struct fields can have `/` character in them (but I could be wrong).

@expede I am not sure I understand your question, what constant value are you referring to ?

Reply:
Oh! Indeed I got my wires crossed. So many specs this week 😅 

> why this called inv here as opposed to ucan/invoke I don't think IPLD struct fields can have / character in them (but I could be wrong).

Yes indeed, it's because of that. I'm not super great with IPLD Schemas yet, but I _think_ I need to add the `(rename "ucan/invoke")` modifier to it.